lib/chef/resource/locale.rb

```diff
@@ -25,67 +25,115 @@ class Locale < Chef::Resource
       description "Use the locale resource to set the system's locale."
       introduced "14.5"
 
+      LC_VARIABLES = %w{LC_ADDRESS LC_COLLATE LC_CTYPE LC_IDENTIFICATION LC_MEASUREMENT LC_MESSAGES LC_MONETARY LC_NAME LC_NUMERIC LC_PAPER LC_TELEPHONE LC_TIME}.freeze
+      LOCALE_CONF = "/etc/locale.conf".freeze
+      LOCALE_REGEX = /\A\S+/.freeze
+
       property :lang, String,
-               default: "en_US.utf8",
-               description: "Sets the default system language."
+               description: "Sets the default system language.",
+               regex: [LOCALE_REGEX],
+               validation_message: "The provided lang not valid. It should be a non-empty string without any leading whitespaces."
```

Comment:
"The provided lang not valid" should be "The provided lang is not valid"

Reply:
Thanks. Updated.